CMSgov/mint-app
# Easi 2611/plan basics to user table

# EASI-2611

NOTE this is dependent on the [generic GQL and Dataloader PR](https://github.com/CMSgov/mint-app/pull/425) and will likely need to be rebased when it is merged.
## Changes and Description

- base task list section to use the user table
    - Generic Functionality to get the user account information for the readyForReviewByUserAccount and readyForClearancebyUserAccount.  
- Audit table updated to use the user table config



## How to test this change
1. Make a few model plans
2. Update a few plan basics, (optionally as different users so you can verify that multiple user Accounts are returned
3. Run this query to get all model plans with the UserAccounts represented by different UserIds
```gql
query modelPlanCollection {
    modelPlanCollection(filter: INCLUDE_ALL) {
        id
        basics{
            modelPlanID
            id      
            createdByUserAccount 
            {
                commonName
            }   
            modifiedByUserAccount
            {
                commonName
            }     
            readyForReviewByUserAccount 
            {
                commonName
            }
            readyForClearanceByUserAccount 
            {
                commonName
            }
        }
        modelName
        archived
        createdBy
        createdDts
        modifiedBy
        modifiedDts
        status
        isFavorite
        isCollaborator
        collaborators {
            createdByUserAccount 
            {
                commonName
                familyName
                givenName
                email
                locale
            }   
            userAccount 
            {
                commonName
                familyName
                givenName
                email
                locale
            }
            modifiedByUserAccount
            {
                commonName
                familyName
                givenName
                email
                locale
            }  
        }

        
        # plan
    }
}
```



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
